server/cluster/shard.go

```diff
@@ -21,3 +24,68 @@ type ShardTablesWithRole struct {
 	tables    []*Table
 	version   uint64
 }
+
+// Shard FSM Const Definition
+// It contains the event name and the state name
+const (
+	EventTransferLeader          = "TransferLeader"
+	EventTransferFollowerFailed  = "TransferFollowerFailed"
+	EventTransferFollower        = "TransferFollower"
+	EventTransferLeaderFailed    = "TransferLeaderFailed"
+	EventPrepareTransferFollower = "PrepareTransferFollower"
+	EventPrepareTransferLeader   = "PrepareTransferLeader"
+	StateLeader                  = "LEADER"
+	StateFollower                = "FOLLOWER"
+	StatePendingLeader           = "PENDING_LEADER"
+	StatePendingFollower         = "PENDING_FOLLOWER"
+)
+
+// Declare the source state array of FSM, avoid creating arrays repeatedly every time you create an FSM
+var (
+	EventTransferLeaderSrc          = []string{StatePendingLeader}
+	EventTransferFollowerFailedSrc  = []string{StatePendingFollower}
+	EventTransferFollowerSrc        = []string{StatePendingFollower}
+	EventTransferLeaderFailedSrc    = []string{StatePendingLeader}
+	EventPrepareTransferFollowerSrc = []string{StateLeader}
+	EventPrepareTransferLeaderSrc   = []string{StateFollower}
+)
+
+// NewFSM /**
+/**
+```
+┌────┐                   ┌────┐
+│ RW ├─────────┐         │ R  ├─────────┐
+├────┘         │         ├────┘         │
+│    Leader    ◀─────────│PendingLeader │
+│              │         │              │
+└───────┬──▲───┘         └───────▲─┬────┘
+        │  │                     │ │
+┌────┐  │  │             ┌────┐  │ │
+│ R  ├──▼──┴───┐         │ R  ├──┴─▼────┐
+├────┘         │         ├────┘         │
+│   Pending    ├─────────▶   Follower   │
+│   Follower   │         │              │
+└──────────────┘         └──────────────┘
+```
+*/
+func NewFSM(role clusterpb.ShardRole) *fsm.FSM {
+	shardFsm := fsm.NewFSM(
+		StateFollower,
+		fsm.Events{
```

Comment:
Actually, not only the `Src` fields can be shared by all the `*fsm.FSM` but also the `fsm.Events`.

Reply:
fix